src/core/prompts/sections/custom-instructions.ts

```diff
@@ -168,8 +172,54 @@ function formatDirectoryContent(dirPath: string, files: Array<{ filename: string
 	)
 }
 
+// kilocode_change start: added functions
+
+async function loadEnabledRulesFromDirectory(
+	rulesDir: string,
+	toggleState: ClineRulesToggles,
+	label: string,
+): Promise<string | null> {
+	if (!(await directoryExists(rulesDir))) {
+		return null
+	}
+
+	const files = await readTextFilesFromDirectory(rulesDir)
+	if (files.length === 0) {
+		return null
+	}
+
+	const rulesContent = await getRuleFilesTotalContent(
+		files.map((f) => f.filename),
+		rulesDir,
+		toggleState,
+	)
+
+	return rulesContent ? `# ${label} from ${rulesDir}:\n${rulesContent}` : null
+}
+
+async function loadEnabledRules(
+	cwd: string,
+	localRulesToggleState: ClineRulesToggles,
+	globalRulesToggleState: ClineRulesToggles,
+): Promise<string> {
+	const globalRulesContent = await loadEnabledRulesFromDirectory(
+		path.join(require("os").homedir(), GlobalFileNames.kiloRules),
```

Comment:
`require("os")` is here because the file sort of claims to be webview compatible (see top of file). I don't think I buy that though.